iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergOutputCommitter.java

```diff
@@ -224,6 +224,11 @@ private void cleanup(JobContext jobContext) throws IOException {
     // If there are reducers, then every reducer will generate a result file.
     // If this is a map only task, then every mapper will generate a result file.
     int expectedFiles = conf.getNumReduceTasks() > 0 ? conf.getNumReduceTasks() : conf.getNumMapTasks();
+    // For Tez, we can only reliably get the number of tasks from the Tez AM, not the job conf
```

Comment:
I wanted to be considerate of MR (even though it's unsupported in upstream Hive), but my initial approach was to not touch it here but set the numReduceTasks to the correct value in the `commitInsertTable`

Reply:
If I understand correctly in Hive3 `commitInsertTable` is not called when the execution engine is MR.
If it would be called anywhere then we are in a big trouble anyway as `HiveIcebergOutputCommitter.commitJob` would be called twice.

So all-in-all I would go for this change

Reply:
Sounds good. Will set the value in the `commitInsertTable`